pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1350,6 +1355,8 @@ func validateChildObjectsInPipelineRunStatus(ctx context.Context, prs v1beta1.Pi
 
 	var err error
 
+	// TODO: Integration test to be introduced for switching in embeddedStatus
+	// https://github.com/tektoncd/pipeline/issues/5999
 	// Verify that we don't populate child references for "full"
```

Comment:
> The errors [here](https://github.com/tektoncd/pipeline/blob/12457e22680fb558720cf4997c9c84568bf27959/pkg/reconciler/pipelinerun/pipelinerun.go#L1353-L1367) don't apply anymore, can you please update this section?

Sorry I haven't fully understood here why does it not apply anymore? I thought though we are solving the bugs for the context that the `embeddedStatus` feature flag is switching these validation shall still apply?

Reply:
The error conditions are that we have "full" embedded-status + childRefs, or "minimal" status with taskruns + runs. With your changes, those conditions can't happen anymore, so I think it would be OK to remove this error.
(However, I could see the argument for not doing so, which is that `validateChildObjects...` shouldn't understand the details of how it's called, so I'll leave this up to you.)

Reply:
Thanks @lbernick  for elaborating. I might leave it until the integration test was implemented to have more thorough testing.